### PR TITLE
[CDAP-19016] Disable emitting spark metrics by default

### PR DESCRIPTION
**Pull request by tivv**
Spark metrics generate metric name dynamically (it includes app name and executor id) and is very resource-consuming to process. We recommend to use spark history server if needed.

### CODE REVIEW

File: cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java [hunk 1]
```diff
@@ -224,7 +224,8 @@ protected void startUp() throws Exception {
       final URI jobFile = context.isPySpark() ? getPySparkScript(tempDir) : createJobJar(tempDir);
       List<File> extraPySparkFiles = new ArrayList<>();
 
-      String metricsConfPath;
+      boolean sparkMetricsEnabled = cConf.getBoolean(Constants.Metrics.SPARK_METRICS_ENABLED);
```
Comment (seanzhougoogle) [inline cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java:227]: would it be better to give it a default value here too ? using the getBoolean(String, boolean) method ?

**Review by seanzhougoogle — APPROVED** [review]
LGTM with minor comments

File: cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java (hunk 1 above)
Comment (tivv) [reply cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java:227]: We usually don't do defaults in get calls and rely on cdap-defaults.xml,. This ensures we have defaults in one place and they don't get out of sync

Comment (masoud-io) [reply cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java:227]: IMO, it's better to define default values in cdap-default as it's been done already.

**Review by masoud-io — APPROVED** [review]

File: cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java (hunk 1 above)
Comment (seanzhougoogle) [reply cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkRuntimeService.java:227]: sg , thanks for the explanation :)

**Pull request merged**